test_regression.py

```diff
@@ -0,0 +1,180 @@
+
+#%%
+import pytest
+from pathlib import Path
+import numpy as np
+import os
+
+from utils import extract_test_geometries, extract_scenario_parameters, execute_and_assert
+
+os.environ["ENDPOINT"] = "https://openeo.dataspace.copernicus.eu/"
+
+
+def test_apply_kernel(auth_connection, tmp_path):
+
+    # Define scenario parameters
+    # Load scenario parameters
+    scenario_name = "apply_spatial_kernel"
+    params = extract_scenario_parameters(scenario_name)
+    
+    # Set up output directory and path
+    output_path = Path(tmp_path) / f"{scenario_name}.nc"
+
+    # Dummy kernel
+    filter_window = np.ones([11, 11])
+    factor = 1 / filter_window.sum()
+
+    # Load collection, and set up progress graph
+    cube = auth_connection.load_collection(
+        collection_id=params['collection_id'],
+        temporal_extent=params['temporal_extent'],
+        spatial_extent=params['spatial_extent'],
+        bands=params['bands']
+    ).apply_kernel(
+        kernel=filter_window,
+        factor=factor)
+    
+    # Excecute and Assert
+    try:
+        execute_and_assert(cube, output_path, scenario_name)
+    except RuntimeError as e:
+        pytest.fail(str(e))
```

Comment:
There is no need to do this try-except with pytest, just let things fail, pytest will catch that properly for you.
Doing it this way even obfuscates interesting information in the pytest report (like full stack trace)